### PR TITLE
DC / MC comparison

### PR DESCRIPTION
**Pull request by dcoeurjo**
using libIGL

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -121,6 +129,115 @@ void doWork()
...
+  //Following libIGL tutorial 715
+  
+  // Indicator function
+  const auto & f_digital = [&](const Eigen::RowVector3d & x)
+  {
+    Point p  ((int)std::round(x[0]),
+              (int)std::round(x[1]),
+              (int)std::round(x[2]));
+    if ((binary_image->domain().isInside(p)) && binary_image->operator()(p))
+      return -1.0;
```
Comment (JacquesOlivierLachaud) [inline main.cpp:180]: Generally an indicatrix is in {0,1} not in {-1,1}. Is this choice important here ?

File: main.cpp [hunk 2]
```diff
@@ -121,6 +129,115 @@ void doWork()
...
+  // Simple finite difference gradients
+  const auto & fd = [](
+                       const std::function<double(const Eigen::RowVector3d&)> & ff,
+                       const Eigen::RowVector3d & x)->Eigen::RowVector3d
+  {
+    const double eps = 1.0 ;
+    Eigen::RowVector3d g;
+    for(int c = 0;c<3;c++)
+    {
+      const Eigen::RowVector3d xp = x+eps*Eigen::RowVector3d(c==0,c==1,c==2);
```
Comment (JacquesOlivierLachaud) [inline main.cpp]: Should be (c==0)  ? 1.0 : 0.0, etc
Here there is the assumption that false is converted to 1.

File: main.cpp [hunk 3]
```diff
@@ -121,6 +129,115 @@ void doWork()
...
+      const double fp = ff(xp);
+      const Eigen::RowVector3d xn = x-eps*Eigen::RowVector3d(c==0,c==1,c==2);
```
Comment (JacquesOlivierLachaud) [inline main.cpp]: Same here

**Review by JacquesOlivierLachaud — APPROVED** [review]
Seems good except the weird conversions c==0 to double.

File: main.cpp (hunk 1 above)
Comment (dcoeurjo) [reply main.cpp:180]: that is the setting for implicit functions <0 inside, >0 outside and DC constructs the 0-iso

File: main.cpp (hunk 2 above)
Comment (dcoeurjo) [reply main.cpp]: dunno.. this is from their tutorial.. I kept that for consistency with their code

Comment (JacquesOlivierLachaud) [reply main.cpp]: Very ugly. I really prefer that you change it to my proposition, which is clear. 
However I did find on site cplusplus.com :
"The conversions from/to bool consider false equivalent to zero (for numeric types) and to null pointer (for pointer types); true is equivalent to all other values and is converted to the equivalent of 1."

Comment (dcoeurjo) [reply main.cpp]: I'll check that

Comment (dcoeurjo) [reply main.cpp]: thx

**Review by JacquesOlivierLachaud — APPROVED** [review]

**Comment by JacquesOlivierLachaud** [comment]
Wait for CI before merging

**Comment by dcoeurjo** [comment]
bots ok
merging, thx

**Pull request merged**